Mathlib/CategoryTheory/GuitartExact.lean

```diff
@@ -0,0 +1,217 @@
+that this square is exact in the sense of Guitart. This means that
+for any `X₃ : C₃`, the induced functor
+`CostructuredArrow L X₃ ⥤ CostructuredArrow R (B.obj X₃)` is final.
+It is also equivalent to the fact that for any `X₂ : C₂`, the
+induced functor `StructuredArrow X₂ T ⥤ StructuredArrow (R.obj X₂) B`
+is initial.
+
+## TODO
+
+* Define the notion of derivability structure from
+[the paper by Kahn and Maltsiniotis][KahnMaltsiniotis2008] using Guitart exact squares
+and construct (pointwise) derived functors using this notion
+
+## References
+* https://ncatlab.org/nlab/show/exact+square
+* [René Guitart, *Relations et carrés exacts*][Guitard1980]
+* [Bruno Kahn and Georges Maltsiniotis, *Structures de dérivabilité*][KahnMaltsiniotis2008]
+
+-/
+
+universe v₁ v₂ v₃ v₄ u₁ u₂ u₃ u₄
+
+namespace CategoryTheory
+
+variable {C₁ : Type u₁} {C₂ : Type u₂} {C₃ : Type u₃} {C₄ : Type u₄}
+  [Category.{v₁} C₁] [Category.{v₂} C₂] [Category.{v₃} C₃] [Category.{v₄} C₄]
+  (T : C₁ ⥤ C₂) (L : C₁ ⥤ C₃) (R : C₂ ⥤ C₄) (B : C₃ ⥤ C₄)
+
+/-- A `2`-square consists of a natural transformation `T ⋙ R ⟶ L ⋙ B`
+involving fours functors `T`, `L`, `R`, `B` that are on the
+top/left/right/bottom sides of a square of categories. -/
+def TwoSquare := T ⋙ R ⟶ L ⋙ B
+
+namespace TwoSquare
+
+variable {T L R B}
+
+@[ext]
+lemma ext (w w' : TwoSquare T L R B) (h : ∀ (X : C₁), w.app X = w'.app X) :
+    w = w' :=
+  NatTrans.ext _ _ (funext h)
+
+variable (w : TwoSquare T L R B)
+
+/-- Given `w : TwoSquare T L R B` and `X₃ : C₃`, this is the obvious functor
+`CostructuredArrow L X₃ ⥤ CostructuredArrow R (B.obj X₃)`. -/
+@[simps! obj map]
+def costructuredArrowRightwards (X₃ : C₃) :
+    CostructuredArrow L X₃ ⥤ CostructuredArrow R (B.obj X₃) :=
+  CostructuredArrow.post L B X₃ ⋙ Comma.mapLeft _ w ⋙
+    CostructuredArrow.pre T R (B.obj X₃)
+
+/-- Given `w : TwoSquare T L R B` and `X₂ : C₂`, this is the obvious functor
+`StructuredArrow X₂ T ⥤ StructuredArrow (R.obj X₂) B`. -/
+@[simps! obj map]
+def structuredArrowDownwards (X₂ : C₂) :
+    StructuredArrow X₂ T ⥤ StructuredArrow (R.obj X₂) B :=
+  StructuredArrow.post X₂ T R ⋙ Comma.mapRight _ w ⋙
+    StructuredArrow.pre (R.obj X₂) L B
+
+section
+
+variable {X₂ : C₂} {X₃ : C₃} (g : R.obj X₂ ⟶ B.obj X₃)
+
+/- In [the paper by Kahn and Maltsiniotis, §4.3][KahnMaltsiniotis2008], given
+`w : TwoSquare T L R B` and `g : R.obj X₂ ⟶ B.obj X₃`, a category `J` is introduced
+and it is observed that it is equivalent to the two categories
+`w.JRightwards g` and `w.JDownwards g`. We shall show below
+that there is an equivalence `w.equivalenceJ g : w.JRightwards g ≌ w.JDownwards g`. -/
+
+/-- Given `w : TwoSquare T L R B` and a morphism `g : R.obj X₂ ⟶ B.obj X₃`, this is the
+category `StructuredArrow (CostructuredArrow.mk g) (w.costructuredArrowRightwards X₃)`,
+see the constructor `JRightwards.mk` for the data that is involved.-/
+abbrev JRightwards :=
+  StructuredArrow (CostructuredArrow.mk g) (w.costructuredArrowRightwards X₃)
```

Comment:
Is there a reason to include `J` in the names, beyond the fact that `J` was used in the paper you cite?

Reply:
I have renamed this one as `StructuredArrowRightwards` and the other one as `CostructuredArrowDownwards`.